modules/ppcp-wc-gateway/services.php

```diff
@@ -2026,4 +2009,44 @@ static function ( ContainerInterface $container ): AuthorizeOrderActionNotice {
 	'button.helper.messages-disclaimers'           => static function ( ContainerInterface $container ): MessagesDisclaimers {
 		return new MessagesDisclaimers();
 	},
+
+	'wcgateway.helper.vaulting-scope'              => static function ( ContainerInterface $container ): bool {
+		try {
+			$token = $container->get( 'api.bearer' )->bearer();
+			return $token->vaulting_available();
+		} catch ( RuntimeException $exception ) {
+			return false;
+		}
+	},
+
+	'button.helper.vaulting-label'                 => static function ( ContainerInterface $container ): string {
+		$vaulting_label = __( 'Enable saved cards and subscription features on your store.', 'woocommerce-paypal-payments' );
+
+		if ( ! $container->get( 'wcgateway.helper.vaulting-scope' ) ) {
+			$vaulting_label .= sprintf(
+				// translators: %1$s and %2$s are the opening and closing of HTML <a> tag.
+				__( ' To use vaulting features, you must %1$senable vaulting on your account%2$s.', 'woocommerce-paypal-payments' ),
+				'<a
+					href="https://docs.woocommerce.com/document/woocommerce-paypal-payments/#enable-vaulting-on-your-live-account"
+					target="_blank"
+				>',
+				'</a>'
+			);
+		}
+
+		$vaulting_label .= '<p class="description">';
+		$vaulting_label .= __( 'This will disable all Pay Later messaging on your site.', 'woocommerce-paypal-payments' );
+		$vaulting_label .= '</p>';
+
+		return $vaulting_label;
+	},
+
+	'button.helper.pay-later-label'                => static function ( ContainerInterface $container ): string {
```

Comment:
@jorgeatorres I wasn't sure about the convention for naming the keys and introducing items in this array of functions. I tried to follow what I saw but I might have missed the point. Glad to update it.

Reply:
There is no clear convention, but since it seems to be related to the settings fields (`wcgateway.settings.fields`), then maybe it should be something like `wcgateway.settings.fields.pay-later-label`.

btw if you are confused by all this container/modularity stuff, this may help: https://gist.github.com/gmazzap/0f9d658c183ff0aaf057a1a9229c3280

Reply:
Makes sense, function name updated.

Thanks for the suggestion and for the reasoning behind the container/modularity approach, @AlexP11223!